cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/inmemory/PrefixedNamespaces.java

```diff
@@ -48,4 +47,23 @@ public static String namespace(CConfiguration cConf, String namespaceId, String
     String namespace = Joiner.on("_").join(rootPrefix, namespaceId);
     return Joiner.on(".").join(namespace, name);
   }
+
+  /**
+   * Extracts the {@link DatasetId} from a dataset's namespaced name.
+   *
+   * @param cConf CDAP configuration
+   * @param namespacedName the dataset's name qualified with a root prefix and a namespace id, as constructed
+   *                       by {@link #namespace(CConfiguration, String, String)}.
+   * @return {@link DatasetId} for the
+   */
+  public static DatasetId getDatasetId(CConfiguration cConf, String namespacedName) {
+    String rootPrefix = cConf.get(Constants.Dataset.TABLE_PREFIX) + "_";
+    Preconditions.checkArgument(namespacedName.startsWith(rootPrefix),
+                                "Expected '%s' to start with '%s'.", namespacedName, rootPrefix);
```

Comment:
would a user understand this message if it were shown in the UI? 

A better message might be: "Unable to determine dataset id from name-spaced table name {namespacedName}, because it does not start with the configured prefix {rootPrefix}".

Feel free to use your own wording.